cf_units/tests/unit/unit/test_Unit.py

```diff
@@ -39,6 +39,12 @@ def test_not_basestring_calendar(self):
         with self.assertRaises(TypeError):
             u = Unit('hours since 1970-01-01 00:00:00', calendar=5)
 
+    def test_hash_replacement(self):
+        hash_unit = 'm # s-1'
+        expected = 'm 1 s-1'
```

Comment:
@duncanwp Is this the behaviour that you were looking for?

Reply:
Yes, this looks great!